samza-core/src/main/scala/org/apache/samza/job/JobRunner.scala

```diff
@@ -92,8 +92,8 @@ class JobRunner(config: Config) extends Logging {
 
     val oldConfig = coordinatorSystemConsumer.getConfig()
     if (resetJobConfig) {
-      info("Deleting old configs that are no longer defined: %s".format(oldConfig.keySet -- config.keySet))
-      (oldConfig.keySet -- config.keySet).foreach(key => {
+      info("Deleting old configs that are no longer defined: %s".format(oldConfig.keySet.asScala -- config.keySet.asScala))
```

Comment:
I've checked `MapConfig` and found that sometimes `MapConfig` class might be backed by immutable Map. So, now I convert it to immutable map and find a diff.